conjurapi/resource.go

```diff
@@ -0,0 +1,25 @@
+package conjurapi
+
+import (
+	"fmt"
+)
+
+func (c *Client) CheckPermission(resourceId, privilege string) (bool, error) {
+	req, err := c.router.CheckPermissionRequest(resourceId, privilege)
+	if err != nil {
+		return false, err
+	}
+
+	resp, err := c.SubmitRequest(req)
+	if err != nil {
+		return false, err
+	}
+
+	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+		return true, nil
+	} else if resp.StatusCode == 404 || resp.StatusCode == 403 {
+		return false, nil
+	} else {
+		return false, fmt.Errorf("Permission check failed with HTTP status %d", resp.StatusCode)
```

Comment:
When would this else be hit?

Reply:
Some other type of server error like 500 or 502